sdks/java/extensions/zetasketch/src/main/java/org/apache/beam/sdk/extensions/zetasketch/HllCount.java

```diff
@@ -0,0 +1,388 @@
+     * bytes-type HLL++ sketch computed from the elements in the input {@code PCollection}.
+     *
+     * <p>Calling {@link Builder#perKey()} returns a {@code PTransform} that takes an input {@code
+     * PCollection<KV<K, byte[]>>} and returns a {@code PCollection<KV<K, byte[]>>} whose contents
+     * is the per-key bytes-type HLL++ sketch computed from the values matching each key in the
+     * input {@code PCollection}.
+     *
+     * <p>Bytes-type sketches cannot be merged with sketches of other types.
+     */
+    public static Builder<byte[]> bytesSketch() {
+      return new Builder<>(HllCountInitFn.forBytes());
+    }
+
+    /**
+     * Builder for the {@code HllCount.Init} combining {@code PTransform}.
+     *
+     * <p>Call {@link #withPrecision(int)} to customize the {@code precision} parameter of the
+     * sketch.
+     *
+     * <p>Call {@link #globally()} or {@link #perKey()} to finalize the {@code PTransform}.
+     *
+     * @param <InputT> element type or value type in {@code KV}s of the input {@code PCollection} to
+     *     the {@code PTransform} being built
+     */
+    public static final class Builder<InputT> {
+
+      private final HllCountInitFn<InputT, ?> initFn;
+
+      private Builder(HllCountInitFn<InputT, ?> initFn) {
+        this.initFn = initFn;
+      }
+
+      /**
+       * Explicitly set the {@code precision} parameter used to compute HLL++ sketch.
+       *
+       * <p>Valid range is between {@value #MINIMUM_PRECISION} and {@value #MAXIMUM_PRECISION}. If
+       * this method is not called, {@value #DEFAULT_PRECISION} will be used. Sketches computed
+       * using different {@code precision}s cannot be merged together.
+       *
+       * @param precision the {@code precision} parameter used to compute HLL++ sketch
+       */
+      public Builder<InputT> withPrecision(int precision) {
+        initFn.setPrecision(precision);
+        return this;
+      }
+
+      /**
+       * Returns a {@code PTransform} that takes an input {@code PCollection<InputT>} and returns a
+       * {@code PCollection<byte[]>} whose contents is the HLL++ sketch computed from the elements
+       * in the input {@code PCollection}.
+       */
+      public PTransform<PCollection<InputT>, PCollection<byte[]>> globally() {
+        return new Globally<>(initFn);
+      }
+
+      /**
+       * Returns a {@code PTransform} that takes an input {@code PCollection<KV<K, InputT>>} and
+       * returns a {@code PCollection<KV<K, byte[]>>} whose contents is the per-key HLL++ sketch
+       * computed from the values matching each key in the input {@code PCollection}.
+       */
+      public <K> PTransform<PCollection<KV<K, InputT>>, PCollection<KV<K, byte[]>>> perKey() {
+        return new PerKey<>(initFn);
+      }
+    }
+
+    private static final class Globally<InputT>
+        extends PTransform<PCollection<InputT>, PCollection<byte[]>> {
```

Comment:
We should provide forwarding for withFanout() and asSingletonView(), as those are both commonly-useful methods on Combine.Globally()

Reply:
Good suggestion! Done.

Reply:
Hi @reuvenlax ! Thanks for taking a look at this. Does the class after changes look good to you. Do you have any comments on the API design, etc.?